WcaOnRails/app/controllers/git_hub_controller.rb

```diff
@@ -0,0 +1,22 @@
+class GitHubController < ApplicationController
+  def edit_translation
+  end
+
+  def update_translation
+    user_login = Octokit.user.login
+    origin_repo = "#{user_login}/worldcubeassociation.org"
+    upstream_repo = "thewca/worldcubeassociation.org"
+    content = params[:translation][:content].gsub(/\r/, '') # We don't want \r characters, but browsers add them automatically.
+    locale = params[:translation][:locale]
+    file_path = "WcaOnRails/config/locales/#{locale}.yml"
+    message = "Update #{locale} translation."
+    content_digest = Digest::SHA1.hexdigest(content)
+    branch_name = "translation-#{content_digest}"
```

Comment:
This makes branch name unique for every single content (unique content). That seems like a reasonable idea for me. It could be any random number though.